Tribler/community/multichain/conversion.py

```diff
@@ -0,0 +1,108 @@
+        :param message: Mesage.impl of SIGNATURE
+        :return: encoding ready to be sent of the network.
+        """
+        payload = message.payload
+        return pack(signature_format, *(payload.up, payload.down, payload.total_up, payload.total_down,
+                                        payload.sequence_number_requester, payload.previous_hash_requester,
+                                        payload.sequence_number_responder, payload.previous_hash_responder)),
+
+    def _decode_signature(self, placeholder, offset, data):
+        """
+        Decode an incoming signature message
+        :param placeholder:
+        :param offset: Start of the SIGNATURE message in the data.
+        :param data: ByteStream containing the message.
+        :return: (offset, SIGNATURE.impl)
+        """
+        if len(data) < offset + signature_size:
+            raise DropPacket("Unable to decode the payload")
+
+        values = unpack_from(signature_format, data, offset)
+        offset += signature_size
+
+        if len(values) != 8:
+            raise DropPacket("Unable to decode the signature")
+
+        up = values[0]
+        if not isinstance(up, int):
+            raise DropPacket("Invalid type up")
+
+        down = values[1]
+        if not isinstance(down, int):
+            raise DropPacket("Invalid type down")
+
+        total_up = values[2]
+        if not isinstance(total_up, int):
+            raise DropPacket("Invalid type total_up")
+
+        total_down = values[3]
+        if not isinstance(total_down, int):
+            raise DropPacket("Invalid type total_down")
+
+        sequence_number_requester = values[4]
+        if not isinstance(sequence_number_requester, int):
+            raise DropPacket("Invalid type sequence_number_requester")
+
+        previous_hash_requester = values[5]
+        if not isinstance(previous_hash_requester, str):
+            raise DropPacket("Invalid type previous_hash_requester")
+
+        sequence_number_responder = values[6]
+        if not isinstance(sequence_number_responder, int):
+            raise DropPacket("Invalid type sequence_number_requester")
+
+        previous_hash_responder = values[7]
+        if not isinstance(previous_hash_responder, str):
+            raise DropPacket("Invalid type previous_hash_responder")
+
+        return \
+            offset, placeholder.meta.payload.implement(up, down, total_up, total_down,
+                                                       sequence_number_requester, previous_hash_requester,
+                                                       sequence_number_responder, previous_hash_responder)
+
+
+def split_function(payload):
+    """
+    This function splits the SIGNATURE MESSAGE in parts.
+    The first to be signed by the requester, and the second the whole message to be signed by the responder
+    :param payload: Encoded message to be split
+    :return: (first_part, whole)
+    """
+    return payload[:request_size], payload
```

Comment:
This is wrong, you're now only signing the first request_size bytes of the payload.
You need to sign everything except the part of Bob, so probably
```
return payload[:-24], payload